samza-core/src/main/java/org/apache/samza/config/StorageConfig.java

```diff
@@ -23,41 +23,48 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
+import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import org.apache.commons.lang3.StringUtils;
 import org.apache.samza.SamzaException;
 import org.apache.samza.execution.StreamManager;
+import org.apache.samza.util.StreamUtil;
 
 
 /**
- * a java version of the storage config
+ * Config helper methods related to storage.
  */
-public class JavaStorageConfig extends MapConfig {
-
+public class StorageConfig extends MapConfig {
   private static final String FACTORY_SUFFIX = ".factory";
+  private static final String CHANGELOG_SUFFIX = ".changelog";
   private static final String STORE_PREFIX = "stores.";
-  private static final String FACTORY = "stores.%s.factory";
-  private static final String KEY_SERDE = "stores.%s.key.serde";
-  private static final String MSG_SERDE = "stores.%s.msg.serde";
-  private static final String CHANGELOG_STREAM = "stores.%s.changelog";
-  private static final String CHANGELOG_SYSTEM = "job.changelog.system";
-  private static final String ACCESSLOG_STREAM_SUFFIX = "access-log";
-  private static final String ACCESSLOG_SAMPLING_RATIO = "stores.%s.accesslog.sampling.ratio";
-  private static final String ACCESSLOG_ENABLED = "stores.%s.accesslog.enabled";
-  private static final int DEFAULT_ACCESSLOG_SAMPLING_RATIO = 50;
-
-  public static final String SIDE_INPUTS = "stores.%s.side.inputs";
-  public static final String SIDE_INPUTS_PROCESSOR_FACTORY = "stores.%s.side.inputs.processor.factory";
-  public static final String SIDE_INPUTS_PROCESSOR_SERIALIZED_INSTANCE = "stores.%s.side.inputs.processor.serialized.instance";
-
-  public JavaStorageConfig(Config config) {
+
+  public static final String FACTORY = "stores.%s" + FACTORY_SUFFIX;
+  public static final String KEY_SERDE = "stores.%s.key.serde";
+  public static final String MSG_SERDE = "stores.%s.msg.serde";
+  public static final String CHANGELOG_STREAM = "stores.%s" + CHANGELOG_SUFFIX;
+  static final String CHANGELOG_SYSTEM = "job.changelog.system";
+  static final String CHANGELOG_DELETE_RETENTION_MS = "stores.%s.changelog.delete.retention.ms";
+  static final long DEFAULT_CHANGELOG_DELETE_RETENTION_MS = TimeUnit.DAYS.toMillis(1);
+  public static final String ACCESSLOG_STREAM_SUFFIX = "access-log";
+  static final String ACCESSLOG_SAMPLING_RATIO = "stores.%s.accesslog.sampling.ratio";
```

Comment:
There doesn't seem to be dependency between the static literals in this class. Should group the public, private and package-protected constants together. Now they're scattered across the class.